radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/LocalSyncService.java

```diff
@@ -0,0 +1,529 @@
+/*
+ * (C) Copyright 2021 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.sync;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.hash.HashCode;
+import com.google.inject.Inject;
+import com.radixdlt.consensus.VerifiedLedgerHeaderAndProof;
+import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.counters.SystemCounters;
+import com.radixdlt.counters.SystemCounters.CounterType;
+import com.radixdlt.crypto.Hasher;
+import com.radixdlt.environment.EventProcessor;
+import com.radixdlt.environment.RemoteEventDispatcher;
+import com.radixdlt.environment.RemoteEventProcessor;
+import com.radixdlt.environment.ScheduledEventDispatcher;
+import com.radixdlt.sync.SyncState.IdleState;
+import com.radixdlt.sync.SyncState.SyncCheckState;
+import com.radixdlt.sync.SyncState.SyncingState;
+import com.radixdlt.ledger.AccumulatorState;
+import com.radixdlt.ledger.DtoCommandsAndProof;
+import com.radixdlt.ledger.LedgerUpdate;
+import com.radixdlt.ledger.LedgerAccumulatorVerifier;
+import com.radixdlt.network.addressbook.AddressBook;
+
+import java.util.Map;
+import java.util.Comparator;
+import java.util.Objects;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.radixdlt.sync.messages.local.LocalSyncRequest;
+import com.radixdlt.sync.messages.local.SyncCheckReceiveStatusTimeout;
+import com.radixdlt.sync.messages.local.SyncCheckTrigger;
+import com.radixdlt.sync.messages.local.SyncRequestTimeout;
+import com.radixdlt.sync.messages.remote.StatusRequest;
+import com.radixdlt.sync.messages.remote.StatusResponse;
+import com.radixdlt.sync.messages.remote.SyncRequest;
+import com.radixdlt.sync.messages.remote.SyncResponse;
+import com.radixdlt.sync.validation.RemoteSyncResponseSignaturesVerifier;
+import com.radixdlt.sync.validation.RemoteSyncResponseValidatorSetVerifier;
+import com.radixdlt.utils.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Processes sync service messages and manages ledger sync state machine.
+ * Thread-safety must be handled by caller.
+ */
+@NotThreadSafe
```

Comment:
If we're gonna keep this file the way it is for now we need to add comments that this class needs to be cleaned up as it is too monolithic.

Reply:
added a comment